c7n/filters/core.py

```diff
@@ -760,3 +760,15 @@ def get_resource_value(self, resource):
         if capture is None:  # regex didn't capture anything
             return None
         return capture.group(1)
+
+
+class StateTransitionFilter(Filter):
+    valid_origin_states = ()
+
+    def filter_resource_state(self, resources, event=None):
+        state_key = self.manager.get_model().state_key
+        states = self.valid_origin_states
+
+        results = [r for r in resources if r[state_key] in states]
```

Comment:
let's log the number of resources that were implicitly filtered out, similar to how some of the other StateTransitionFilters work, e.g. https://github.com/cloud-custodian/cloud-custodian/blob/c190c8672c6f4a035a34865ddc2aa399076460e1/c7n/resources/sagemaker.py#L328-L329